include/pybind11/pybind11.h

```diff
@@ -35,7 +43,9 @@
 #  pragma GCC diagnostic ignored "-Wunused-but-set-parameter"
 #  pragma GCC diagnostic ignored "-Wunused-but-set-variable"
 #  pragma GCC diagnostic ignored "-Wmissing-field-initializers"
-#  pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#  if !PYBIND11_HAS_STD_LAUNDER
+#    pragma GCC diagnostic ignored "-Wstrict-aliasing"
+#  endif
```

Comment:
In case we don't have `std::launder`, we can't tackle `-Wstrict-aliasin`. At least not easily, without breaking or a major refactoring.